### ISSUE TITLE
[Bug]: Accessibility of Addons overflow tab lacking

### ISSUE THREAD

**Issue by dgibson666** [storybookjs/storybook#23257]
### Describe the bug

When the Addons panel is narrow enough that some tabs get rolled up into a dropdown (very easy when docking it on the right), those overflow tabs are completely inaccessible via keyboard. The "Addons" tab toggles the menu open/closed, but neither the tab key nor arrows will navigate to those expanded options. The button representing this tab also has no accessibility information about the menu such as `aria-expanded` or `aria-haspopup`, which are useful to screen reader users.

### To Reproduce

* Run storybook
* Dock the addons to the right
* Shrink the addons panel so that not all tabs fit horizontally and some get rolled up under an "Addons (downward triangle)" tab.
* Click the first tab in the addons panel
* Press the Tab key to navigate to the "Addons" tab at the end.
* Press Enter/space to activate it
* Press tab or down arrow to attempt to navigate the items in the menu - neither will reach them.

### System

```shell
Environment Info:

  System:
    OS: Windows 10 10.0.22621
    CPU: (24) x64 13th Gen Intel(R) Core(TM) i7-13700F
  Binaries:
    Node: 20.2.0 - C:\Program Files\nodejs\node.EXE
    npm: 9.6.6 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: Spartan (44.22621.1848.0), Chromium (114.0.1823.58)
```

**Comment by rasvii** [comment]
Hi @vanessayuenn, I am interested in working on this issue. Could you please assign it to me?

**Comment by dgibson666** [comment]
I just upgraded to Storybook 8 and verified this issue is still present.

**Comment by vanessayuenn** [comment]
@rasvii we don't reserve `good first issues` but please feel free to work on it and submit a PR! Be sure to check the [contribution guide](https://storybook.js.org/docs/react/contribute/how-to-contribute) and hop onto the [Storybook Discord](https://discord.gg/storybook) if you need any help getting started. Looking forward to your contribution! ✨

**Comment by Sidnioulz** [comment]
This was fixed in the 10.1 toolbar rework!
